qadence/backends/api.py

```diff
@@ -14,67 +21,47 @@ def backend_factory(
     configuration: BackendConfiguration | dict | None = None,
 ) -> Backend | DifferentiableBackend:
     backend_inst: Backend | DifferentiableBackend
-    backends = available_backends()
     try:
-        backend_name = BackendName(backend)
-    except ValueError:
-        raise NotImplementedError(f"The requested backend '{backend}' is not implemented.")
-    try:
-        BackendCls = backends[backend_name]
-    except Exception as e:
-        raise ImportError(
-            f"The requested backend '{backend_name}' is either not installed\
-              or could not be imported due to {e}."
-        )
-
-    default_config = BackendCls.default_configuration()
-    if configuration is None:
-        configuration = default_config
-    elif isinstance(configuration, dict):
-        configuration = config_factory(backend_name, configuration)
-    else:
-        # NOTE: types have to match exactly, hence we use `type`
-        if not isinstance(configuration, type(BackendCls.default_configuration())):
-            raise ValueError(
-                f"Given config class '{type(configuration)}' does not match the backend",
-                f" class: '{BackendCls}'. Expected: '{type(BackendCls.default_configuration())}.'",
-            )
-
-    # Create the backend
-    backend_inst = BackendCls(
-        config=configuration
-        if configuration is not None
-        else BackendCls.default_configuration()  # type: ignore[attr-defined]
-    )
+        BackendCls = import_backend(backend)
+        default_config = BackendCls.default_configuration()
+        if configuration is None:
+            configuration = default_config
+        elif isinstance(configuration, dict):
+            configuration = config_factory(backend, configuration)
+        else:
+            # NOTE: types have to match exactly, hence we use `type`
+            if not isinstance(configuration, type(BackendCls.default_configuration())):
+                expected_cfg = BackendCls.default_configuration()
+                raise ValueError(
+                    f"Given config class '{type(configuration)}' does not match the backend",
+                    f" class: '{BackendCls}'. Expected: '{type(expected_cfg)}.'",
+                )
 
+        # Instantiate the backend
+        backend_inst = BackendCls(  # type: ignore[operator]
+            config=configuration
```

Comment:
Is this check necessary ? It feels that at this point, we've exhausted all possibilities for `configuration` and excluded cases where it is undefined.